queries.sql

```diff
@@ -6,4 +6,66 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * from animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+BEGIN TRANSACTION;
+
+UPDATE animals 
+SET species = 'unspecified';
+
+SELECT * FROM animals
+WHERE species = 'unspecified';
+
+Rollback;
+
+SELECT * FROM animals;
+
+begin;
+
+update animals
+set species = 'digimon'
+where name like '%mon';
+
+update animals 
+set species = 'pokemon'
+where species is null;
+
+select * from animals 
+
+commit;
+
+select * from animals;
+
+begin;
+
+delete from animals;
+
+select * from animals;
+
+rollback;
+
+select * from animals;
+
+begin;
+
+delete from animals
+where date_of_birth > '2022-01-01';
+
+savepoint sp1;
+
+update animals 
+set weight_kg = weight_kg * -1;
+
+commit;
+select * from animals
```

Comment:
- [ ] It seems you missed one crucial aspect in this transaction, after you update the animal's weight in line `56` you should rollback to the savepoint and perform another update for animals whose weight is less than 0 before making any commit.